experimental/devices/hd44780/hd44780.go

```diff
@@ -0,0 +1,256 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+// Package hd44780 controls the Hitachi LCD display chipset HD-44780
+//
+// Datasheet
+//
+// https://www.sparkfun.com/datasheets/LCD/HD44780.pdf
+package hd44780
+
+import (
+	"fmt"
+	"time"
+
+	"periph.io/x/periph/conn"
+	"periph.io/x/periph/conn/gpio"
+)
+
+// lineTwo offset for the second line in the LCD buffer.
+const lineTwo = 0x40
+
+// Dev is the 4-bit addressing device for HD-44780
+type Dev struct {
+	// data pins
+	dataPins []gpio.PinOut
+
+	// register select pin
+	rsPin gpio.PinOut
+
+	// enable pin
+	enablePin gpio.PinOut
+}
+
+// New creates and initializes the LCD device
+//	data - references to data pins
+//	rs - rs pin
+//	e - strobe pin
+func New(data []gpio.PinOut, rs, e gpio.PinOut) (*Dev, error) {
+	if len(data) != 4 {
+		return nil, fmt.Errorf("expected 4 data pins, passed %s", len(data))
+	}
+	dev := &Dev{
+		dataPins:  data,
```

Comment:
I'm wondering if we should play defensive here and create a local slice. Otherwise if the caller alter data after this function call, it messes up the code here. A way to do this is to replace line 26 as an array instead:

```
dataPins [4]gpio.PinOut
```

and do

```
copy(d.dataPins, data)
```